src/components/link-preview/wikipedia.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+
+import cachedFetch from './cached-fetch';
+import ScrollSafe from './scroll-helpers/scroll-safe';
+import FoldableContent from './scroll-helpers/foldable-content';
+
+const WIKIPEDIA_RE = /^https?:\/\/(\w+)\.wikipedia\.org\/wiki\/([^\/]+)/i;
+
+export function canShowURL(url) {
+  return WIKIPEDIA_RE.test(url);
+}
+
+const initialState = {
+  previewData: null,
+};
+
+class WikipediaPreview extends React.Component {
+  state = {...initialState};
+
+  async updatePreview() {
+    const [, lang, term] = WIKIPEDIA_RE.exec(this.props.url);
+    const data = await cachedFetch(`https://${lang}.wikipedia.org/api/rest_v1/page/summary/${term}`);
+    if (data.error) {
+      this.setState({previewData: null});
+    } else {
+      this.setState({previewData: data});
+    }
+  }
+
+  componentDidMount() {
+    this.updatePreview();
+  }
+
+  componentDidUpdate(newProps) {
```

Comment:
Actually it's even better to use `componentWillReceiveProps`. I'll make a change.